src/ripple/app/misc/impl/AMMUtils.cpp

```diff
@@ -287,17 +291,25 @@ initializeFeeAuctionVote(
     Issue const& lptIssue,
     std::uint16_t tfee)
 {
+    auto const& rules = view.rules();
     // AMM creator gets the voting slot.
     STArray voteSlots;
-    STObject voteEntry{sfVoteEntry};
+    STObject voteEntry = STObject::makeInnerObject(sfVoteEntry, rules);
     if (tfee != 0)
         voteEntry.setFieldU16(sfTradingFee, tfee);
     voteEntry.setFieldU32(sfVoteWeight, VOTE_WEIGHT_SCALE_FACTOR);
     voteEntry.setAccountID(sfAccount, account);
     voteSlots.push_back(voteEntry);
     ammSle->setFieldArray(sfVoteSlots, voteSlots);
     // AMM creator gets the auction slot for free.
-    auto& auctionSlot = ammSle->peekFieldObject(sfAuctionSlot);
+    // AuctionSlot is created on AMMCreate and updated on AMMDeposit
+    // when AMM is in an empty state
+    if (!ammSle->isFieldPresent(sfAuctionSlot))
```

Comment:
This needs to be protected by an an amendment check. In general, when adding transaction breaking code, the code should follow the pattern:
```
if (amendment_on){
// do the new thing
}
else{
// do the old thing
}
```
It makes the code much easier to audit and makes it much less likely that transaction breaking changes will sneak in.